### PR TITLE
api: exclude sec-approval from the reviewers list (bug 1590225)

### PR DESCRIPTION
**Pull request by mars-f**
Exclude the sec-approval team from the list of code reviewers when displaying a
commit message in the UI and when building a commit message to send to the
transplant service.  This allows the sec-approval team to consult on a change 
without revealing the security-sensitive nature of the patch in the public
commit message.

### CODE REVIEW

File: landoapi/api/stacks.py [hunk 1]
```diff
@@ -97,6 +102,12 @@ def get(revision_id):
...
         reviewers = get_collated_reviewers(revision)
+
+        # The sec-approval group must not appear in the commit message reviewers list.
+        # Bug 1590225
+        if sec_approval_project_phid in reviewers:
+            del reviewers[sec_approval_project_phid]
+
         accepted_reviewers = [
             reviewer_identity(phid, users, projects).identifier
             for phid, r in reviewers.items()
```
Comment (mars-f) [inline landoapi/api/stacks.py]: @smacleod I'm concerned by the code duplication here.  The same ~15 lines of code to gather contextual information to build a commit message appears in `lando.api.stacks` and `lando.api.transplants`.  Is it time to refactor this into a common routine?

Comment (smacleod) [reply landoapi/api/stacks.py]: I'd prefer if you leave as-is for now, there's other changes going on in the area and it's not a huge duplication at this point.

File: landoapi/api/stacks.py [hunk 2]
```diff
@@ -97,6 +102,12 @@ def get(revision_id):
         summary = PhabricatorClient.expect(fields, "summary")
         bug_id = get_bugzilla_bug(revision)
         reviewers = get_collated_reviewers(revision)
+
+        # The sec-approval group must not appear in the commit message reviewers list.
+        # Bug 1590225
+        if sec_approval_project_phid in reviewers:
+            del reviewers[sec_approval_project_phid]
```
[the remark this answers is not in the document]
Comment (mars-f) [reply landoapi/api/stacks.py]: Fixed

File: tests/test_transplants.py [hunk 3]
```diff
@@ -751,6 +753,39 @@ def test_integrated_transplant_revision_with_unmapped_repo(
...
+        },
+        headers=auth0_mock.mock_headers,
+    )
+    assert response == 202
+
+    # Check the transplanted patch for our alternate commit message.
+    patch = s3.Object(
+        app.config["PATCH_BUCKET_NAME"], patches.name(revision["id"], diff["id"])
+    )
+    patch = PatchHelper(BytesIO(patch.get()["Body"].read()))
```
[the remark this answers is not in the document]
Comment (mars-f) [reply tests/test_transplants.py]: Fixed

**Pull request merged**